io/disk/disktest.py

```diff
@@ -0,0 +1,115 @@
+import os
+import sys
+import subprocess
+import logging
+
+
+from avocado import Test
+from avocado import main
+from avocado.utils import memory
+from avocado.utils import disk
+from avocado.utils import process
+from avocado.utils import build
+from avocado.utils.software_manager import SoftwareManager
+
+
+class Disktest(Test):
+
+    """
+    Avocado module for disktest.
+    Pattern test of the disk, using unique signatures for each block and each
+    iteration of the test. Designed to check for data corruption issues in the
+    disk and disk controller.
+    It writes 50MB/s of 500KB size ops.
+    """
+    version = 2
+
+    def setUp(self):
+        """
+        Verifies if we have gcc to compile disktest.
+        """
+        softm = SoftwareManager()
+        if not softm.check_installed("gcc") and not softm.install("gcc"):
+            self.error('Gcc is needed for the test to be run')
+        self.disks = self.params.get('disks', default=None)
+        self.gigabytes = self.params.get('gigabytes', default=None)
+        self.chunk_mb = self.params.get('chunk_mb', default=None)
+        self.memory_mb = memory.memtotal() / 1024 / 8
+        self.data_dir = os.path.join(os.path.abspath(self.datadir))
```

Comment:
I guess you forgot to put something here? Btw the `datadir` is already abspath, so you can probably directly use `self.datadir`.